src/main/java/org/opensearch/plugin/insights/core/service/categorizer/IndicesFieldTypeCache.java

```diff
@@ -59,9 +59,33 @@ public Iterable<Index> keySet() {
         return cache.keys();
     }
 
+    /**
+     * Estimated memory consumption of the cache in bytes
+     */
+    public Long getEstimatedSize() {
+        long totalWeight = 0;
+
+        // Iterate over the keys of the cache
+        for (Index index : cache.keys()) {
+            // Get the corresponding IndexFieldMap
+            IndexFieldMap indexFieldMap = cache.get(index);
+
+            // Ensure the value is not null before calling weight()
+            if (indexFieldMap != null) {
+                totalWeight += indexFieldMap.weight();
+            }
+        }
+
+        return totalWeight;
+    }
```

Comment:
Shouldn't we compute this while adding or removing from cache?

Reply:
Yes, that makes sense since frequency of `health_stats` API calls will be greater than frequency of items going in/out of this cache.